sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -482,11 +482,17 @@ case class Sort(
 
 /** Factory for constructing new `Range` nodes. */
 object Range {
-  def apply(start: Long, end: Long, step: Long, numSlices: Option[Int]): Range = {
+  def apply(start: Long, end: Long, step: Long, numSlices: Option[Int])
+      : LeafNode with MultiInstanceRelation = {
     val output = StructType(StructField("id", LongType, nullable = false) :: Nil).toAttributes
-    new Range(start, end, step, numSlices, output)
+    if (start == end || (start < end ^ 0 < step)) {
+      LocalRelation(output)
```

Comment:
Thank you for review, @cloud-fan . In that case, which spec can we use for that empty `Range` which is returned from `Range.empty`? For the following cases, it seems that we need to remove one of the following precondition again.
```
  require(step != 0, s"step ($step) cannot be 0")
  require(start != end, s"start ($step) cannot be equal to end ($end)")
  require(start < end ^ step < 0, s"the sign of step ($step) is invalid for range ($start, $end)")
```